MarkupGatewayPlugin.inc.php

```diff
@@ -436,15 +436,27 @@ function _process($submissionFile, $stage, $target) {
 		switch ($target) {
 			case 'xml-conversion':
 				$params = array(
-					'stage' => $stage,
-					'assocType' => (int)$submissionFile->getAssocType(),
-					'assocId' => (int)$submissionFile->getAssocId(),
-					'UserGroupId' => (int)$submissionFile->getUserGroupId()
+					'stage' 	=> $stage,
+					'assocType' 	=> (int)$submissionFile->getAssocType(),
+					'assocId' 	=> (int)$submissionFile->getAssocId(),
+					'UserGroupId' 	=> (int)$submissionFile->getUserGroupId()
 				);
 				$this->addXmlDocumentToFileList($submission, "{$extractionPath}/document.xml", $params);
 				break;
 
 			case 'galley-generate':
+				// Always populate production ready files with xml document.
+				$xmlFileName = "document" . '__' . date('Y-m-d_h:i:s') .'.xml';
+				$params = array(
+					'stage' 	=> SUBMISSION_FILE_PRODUCTION_READY,
+					'assocType' 	=> (int)$submissionFile->getAssocType(),
+					'assocId' 	=> 0,
```

Comment:
Should `assocId` and `userGroupId` really be `0`? Shouldn't `UserGroupId` be `userGroupId`?

Reply:
@axfelix @asmecher : what should be the value of assocId? 
@asmecher : Sorry I was going to ask you. I couldn't figure out how to determine current user's group. Could you point me to a piece of code doing that?

Reply:
@kaschioudi, usually users will only have one possible user group, so the `userGroupId` will be automatically chosen accordingly. In the rare case there are multiple options, there will be a drop-down on the upload form allowing them to choose. This is implemented in `lib/pkp/controllers/wizard/fileUpload/form/SubmissionFilesUploadForm.inc.php` and `lib/pkp/templates/controllers/wizard/fileUpload/form/fileUploadForm.tpl` (look for `uploaderUserGroupOptions`).

Reply:
@asmecher  

I'm reusing the assocType associated to the submission file being converted. Should I do the same for assocId? Otherwise what would be correct values? 

Regarding user's group id, since I can't really display a drop-down, is it okay to pick the first userGroupId (`$userGroup = $userGroups->next();`) since as per your comment above users will have one user group.

Reply:
@asmecher : https://github.com/kaschioudi/ojs3-markup/commit/21cbe28a963eb12aba3a06acef668f15d8a30463

Does this make sense?

Reply:
Yes, picking the first user group makes sense to me.

Reply:
For production-ready files, assoc type and assoc ID should both be unset (null).